File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.py
def solution(myString):

    return [len(str) for str in myString.split('x')]

# 다른 풀이
return list(map(lambda x: len(x), myString.split('x')))
# map(function, iterable, ...) : 주어진 함수를 시퀀스(리스트, 튜플 등)의 각 요소에 적용하여 새로운 시퀀스를 반환
#     - function : 시퀀스의 각 요소에 적용할 함수
#     - iterable : 입력 시퀀스

File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(String myString) {
        
        return Arrays.stream(myString.split("x", myString.length()))
              .mapToInt(str -> str.length()) // .mapToInt(String::length)
              .toArray();
    }
}

// 다른 풀이
return Arrays.stream(myString.split("x", -1)).map(String::length).toArray(Integer[]::new);
// split(String regex, int limit) : limit을 -1로 전달하면 분할된 문자열에 제한 없이 모두 배열에 포함
// mapToInt() : IntStream 반환
// map() : Stream<Integer> 반
